libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -559,11 +596,41 @@ bool AP_Follow::have_target(void) const
     }
 
     // check for timeout
-    if ((_last_location_update_ms == 0) || (AP_HAL::millis() - _last_location_update_ms > AP_FOLLOW_TIMEOUT_MS)) {
+    if ((_last_location_update_ms == 0) || (AP_HAL::millis() - _last_location_update_ms > (uint32_t)_timeout_ms)) {
+        return false;
+    }
+    return true;
+}
+
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane)
+// create a single method to retrieve all the relevant values in one shot for Lua
+/* replaces the following Lua calls
+   target_distance, target_distance_offsets, target_velocity = follow:get_target_dist_and_vel_ned()
+   target_location, target_velocity = follow:get_target_location_and_velocity()
+   target_location_offset, target_velocity = follow:get_target_location_and_velocity_ofs()
+   local xy_dist = follow:get_distance_to_target() -- this value is set by get_target_dist_and_vel_ned() - why do I have to know this?
+   local target_heading = follow:get_target_heading_deg()
+*/
+bool AP_Follow::get_target_info(Vector3f &dist_ned, Vector3f &dist_with_offs,
+                                Vector3f &target_vel_ned, Vector3f &target_vel_ned_ofs,
+                                Location &target_loc, Location &target_loc_ofs,
+                                float &target_dist_ofs
+                                )
+{
+    // The order here is VERY important. This needs to called first because it updates a lot of internal variables
```

Comment:
If the ordering is important does that mean the existing bindings make it possible for lua users to get it wrong?

The lua calls really should not be changing the internal follow state. Maybe if we fixed that this new method would not be needed?

Reply:
> If the ordering is important does that mean the existing bindings make it possible for lua users to get it wrong?

Absolutely yes.

> The lua calls really should not be changing the internal follow state. Maybe if we fixed that this new method would not be needed?

I think that's a good idea, but outside the scope of this PR? Happy to refactor later.

Reply:
If we did the refactor would that remove the need for the new method?

Reply:
Well it still leaves the case that if you call multiple methods Lua might get values from different loops because of the lower priority scheduling in Lua. One method means you are guaranteed to get a consistent set of values. 

One method is also a lower flash cost for the binding I think.